rasa_nlu/utils/mitie_utils.py

```diff
@@ -1,38 +1,45 @@
-from __future__ import unicode_literals
-from __future__ import print_function
-from __future__ import division
 from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import os
+
 from builtins import str
 from typing import Optional
+from typing import Text
 
 from rasa_nlu.components import Component
 from rasa_nlu.model import Metadata
 
 
 class MitieNLP(Component):
-
     name = "init_mitie"
 
     context_provides = {
         "pipeline_init": ["mitie_feature_extractor"],
     }
 
-    def __init__(self, extractor=None):
+    def __init__(self, mitie_file, extractor=None):
         self.extractor = extractor
+        self.mitie_file = mitie_file
+        MitieNLP.ensure_proper_language_model(self.extractor)
+
+    @classmethod
+    def create(cls, mitie_file):
+        import mitie
+        extractor = mitie.total_word_feature_extractor(mitie_file)
+        return MitieNLP(mitie_file, extractor)
 
     @classmethod
     def cache_key(cls, model_metadata):
-        # type: (Metadata) -> str
+        # type: (Metadata) -> Optional[Text]
 
-        return cls.name + "-" + str(model_metadata.metadata.get("mitie_feature_extractor_fingerprint"))
+        mitie_file = model_metadata.metadata.get("mitie_file", None)
+        return cls.name + "-" + str(os.path.abspath(mitie_file)) if mitie_file else None
```

Comment:
Concatenating string and NoneType breaks in both python2 and 3. `""` works if this is what you want